paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/PaymentSheetTest.kt

```diff
@@ -204,6 +204,66 @@ internal class PaymentSheetTest {
         assertThat(countDownLatch.await(5, TimeUnit.SECONDS)).isTrue()
     }
 
+    @Test
+    fun testDeferredIntentFailedCardPayment() {
+        networkRule.enqueue(
+            method("GET"),
+            path("/v1/elements/sessions"),
+        ) { response ->
+            response.testBodyFromFile("elements-sessions-deferred_payment_intent.json")
+        }
+
+        val activityScenarioRule = composeTestRule.activityRule
+        val scenario = activityScenarioRule.scenario
+        scenario.moveToState(Lifecycle.State.CREATED)
+        lateinit var paymentSheet: PaymentSheet
+        scenario.onActivity {
+            PaymentConfiguration.init(it, "pk_test_123")
+            paymentSheet = PaymentSheet(
+                activity = it,
+                createIntentCallback = {
+                    CreateIntentCallback.Result.Failure(
+                        cause = Exception("We don't accept visa"),
+                        displayMessage = "We don't accept visa"
+                    )
+                }
+            ) {
+                // No PaymentSheetResult callback
```

Comment:
Can we throw here instead? Since we don't expect it to be called.

Reply:
Good call

Reply:
Done